src/Components/Aspire.Npgsql/AspirePostgreSqlNpgsqlExtensions.cs

```diff
@@ -99,14 +101,19 @@ public static void AddKeyedNpgsqlDataSource(this IHostApplicationBuilder builder
         }
     }
 
-    private static void RegisterNpgsqlServices(this IHostApplicationBuilder builder, NpgsqlSettings settings, string configurationSectionName, string connectionName, object? serviceKey)
+    private static void RegisterNpgsqlServices(this IHostApplicationBuilder builder, NpgsqlSettings settings, string configurationSectionName, string connectionName, object? serviceKey, Action<NpgsqlDataSourceBuilder>? configureDataSourceBuilder)
     {
         if (serviceKey is null)
         {
             // delay validating the ConnectionString until the DataSource is requested. This ensures an exception doesn't happen until a Logger is established.
             builder.Services.AddNpgsqlDataSource(settings.ConnectionString ?? string.Empty, dataSourceBuilder =>
             {
                 ValidateConnection();
+
+                if (configureDataSourceBuilder is not null)
+                {
+                    configureDataSourceBuilder(dataSourceBuilder);
+                }
```

Comment:
```suggestion
                configureDataSourceBuilder?.Invoke(dataSourceBuilder);
```

nit: I believe people that syntax in this repos